services/requester/requester.go

```diff
@@ -597,7 +526,111 @@ func (e *EVM) dryRunTx(
 	return result, nil
 }
 
-func AddOne64th(n uint64) uint64 {
-	// NOTE: Go's integer division floors, but that is desirable here
-	return n + (n / 64)
+// buildTransaction creates a flow transaction from the provided script,
+// with the arguments and signs it with the configured COA account.
+func (e *EVM) buildTransaction(
+	ctx context.Context,
+	script []byte,
+	args ...cadence.Value,
+) (*flow.Transaction, error) {
+	// building and signing transactions should be blocking, so we don't have keys conflict
+	e.mux.Lock()
+	defer e.mux.Unlock()
+
+	var (
+		g           = errgroup.Group{}
+		err1, err2  error
+		latestBlock *flow.Block
+		account     *flow.Account
+	)
+	// execute concurrently so we can speed up all the information we need for tx
+	g.Go(func() error {
+		latestBlock, err1 = e.client.GetLatestBlock(ctx, true)
+		return err1
+	})
+	g.Go(func() error {
+		account, err2 = e.client.GetAccount(ctx, e.config.COAAddress)
```

Comment:
**Note:** Having to get the account for each call of `eth_sendRawTransaction` might be sub-optimal. The only reason we need this is because we used to have a metric for the operator's balance:
```go
e.collector.OperatorBalance(account)
```
which is called below. Maybe it's better to move this metric somewhere else.

Reply:
Update: I've actually used that `account` variable, to fetch the proper `SequenceNumber` when using ` accKey.SetProposerPayerAndSign(flowTx, account)`.